### DISCUSSION TITLE
MenuItem hotkeys are not available until the contextMenu appears.

### DISCUSSION THREAD

**Discussion by alvahtin** [AvaloniaUI/Avalonia#15319] [Q&A] [answered]
This is bug? Or needed something?

I'm using ContextMenu with MenuItems, which has hotkeys. While the context menu has never been opened, the hotkeys do not work.

I looked at the Visual Tree in Avalonia DevTools, and saw that, before the first opening of the context menu, the ActualThemeVariant property is null. But after opening it has any value.

![image](https://github.com/AvaloniaUI/Avalonia/assets/50268307/01da0df1-915c-4cb5-89f8-c36d4a336c21)

![image](https://github.com/AvaloniaUI/Avalonia/assets/50268307/99b7542d-3abe-4ac1-a283-92b2722273b7)

As I understand it, the context menu is not fully loaded before the first opening. Is there a way to make it load completely?

**Comment by workgroupengineering** [comment, marked as answer]
```xaml
<Window xmlns="https://github.com/avaloniaui"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        ....
     >
  <Window.KeyBindings>
    <KeyBinding Gesture="F8" Command="{Binding YourCommand}"/>
  </Window.KeyBindings>
  <l:DataGridExt ...>
  </l:DataGridExt ...>
</Window>
```

**Reply by alvahtin** [reply]
Thank you!

**Comment by jp2masa** [comment]
It's also an open issue: #2441.
